React/Views/RCTMaskedView.m

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "RCTMaskedView.h"
+
+@implementation RCTMaskedView
+
+- (void) layoutSubviews
+{
+  [super layoutSubviews];
+  
+  // When a mask is set on this view, we need to make sure
+  // the mask view isn't in the view hierarchy (iOS won't use
+  // the view as a mask if it exists in the view hierarchy).
+  // Then reset the `maskView` property to make sure that
+  // the mask is set correctly.
+  if (self.maskView != nil) {
+    if (self.maskView.superview != nil) {
+      UIView *_maskView = self.maskView;
```

Comment:
`maskView` is set when the prop is set, but at that point, the view we're setting as a mask is still a part of the view hierarchy, so setting it doesn't do what we want it to do -- we need to reset it in order for the CALayer mask to be applied correctly.

Honestly, as I type this I realize there's a clearer way to write this while maintaining the same behavior, so gonna do that.